Source/WebCore/style/StyleAdjuster.cpp

```diff
@@ -583,6 +583,12 @@ void Adjuster::adjust(RenderStyle& style, const RenderStyle* userAgentAppearance
 
         style.setEventListenerRegionTypes(computeEventListenerRegionTypes(m_document, style, *m_element, m_parentStyle.eventListenerRegionTypes()));
 
+#if ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)
+        // Every element will automatically get an interaction region which is not useful, ignoring the `cursor: pointer;` on the body.
+        if (is<HTMLBodyElement>(*m_element) && style.cursor() == CursorType::Pointer && style.eventListenerRegionTypes().contains(EventListenerRegionType::MouseClick))
+            style.setCursor(CursorType::Auto);
```

Comment:
I'm not sure we should be *actually* adjusting the style like this, should we? Is it possible to either identify this later (in InteractionRegion), to keep the heuristic all in one place, or to just remove the bit from `eventListenerRegionTypes` instead of actually mutating the style in a visible way?

Reply:
We can go with this for now since nobody came up with a better plan.